chrome/content/zotero/zoteroPane.js

```diff
@@ -1243,39 +1243,41 @@ var ZoteroPane = new function()
 			document.getElementById('zotero-items-tree').view = this.itemsView;
 			
 			try {
+				let tree = document.getElementById('zotero-items-tree');
 				let treecols = document.getElementById('zotero-items-columns-header');
 				let treecolpicker = treecols.boxObject.firstChild.nextSibling;
 				let menupopup = treecolpicker.boxObject.firstChild.nextSibling;
 				// Add events to treecolpicker to update menu before showing/hiding
 				let attr = menupopup.getAttribute('onpopupshowing');
 				if (attr.indexOf('Zotero') == -1) {
-					menupopup.setAttribute('onpopupshowing', 'ZoteroPane.itemsView.onColumnPickerShowing(event);')
+					menupopup.setAttribute('onpopupshowing', 'ZoteroPane.itemsView.onColumnPickerShowing(event); '
 						// Keep whatever else is there
-						+ ' ' + attr;
-					menupopup.setAttribute('onpopuphidden', 'ZoteroPane.itemsView.onColumnPickerHidden(event);')
+						+ attr);
+					menupopup.setAttribute('onpopuphidden', 'ZoteroPane.itemsView.onColumnPickerHidden(event); '
 						// Keep whatever else is there
-						+ ' ' + menupopup.getAttribute('onpopuphidden');
+						+ menupopup.getAttribute('onpopuphidden'));
 				}
 				
-				// Hide certain columns for feeds
-				let elems = treecols.getElementsByAttribute('hidden-in', '*');
-				for (let i = 0; i < elems.length; i++) {
-					let shouldHide = elems[i].getAttribute('hidden-in').split(' ')
-						.indexOf(collectionTreeRow.type) != -1;
-					let userHidden = elems[i].hasAttribute('user-hidden');
-					// The slightly convoluted logic here is such:
-					// If the current tree being displayed is in the list of types 'hidden-in'
-					// specifies and it's not already been hidden by another 'hidden-in' rule
-					// (i.e. its normal state of being displayed has not been suppressed yet),
-					// then hide it
-					if (shouldHide && !userHidden) {
-						// Store old value on entry to feed view
-						elems[i].setAttribute('user-hidden', elems[i].getAttribute('hidden'));
-						elems[i].setAttribute('hidden', 'true');
-					} else if (!shouldHide && userHidden) {
-						// Restore old value on exit from feed view
-						elems[i].setAttribute('hidden', elems[i].getAttribute('user-hidden'));
-						elems[i].removeAttribute('user-hidden');
+				// Items view column visibility for different groups
+				let prevViewGroup = tree.getAttribute('current-view-group');
+				let curViewGroup = collectionTreeRow.visibilityGroup;
+				tree.setAttribute('current-view-group', curViewGroup);
+				if (curViewGroup != prevViewGroup) {
+					let cols = Array.prototype.slice.call(treecols.querySelectorAll('treecol'));
```

Comment:
`Array.from()`, and though it almost certainly doesn't matter, might as well use the (generally faster) `treecols.getElementsByTagName()`